src/Thentos/Backend/Api/Adhocracy3.hs

```diff
@@ -324,32 +324,39 @@ sendUserConfirmationMail smtpConfig user callbackUrl =
 activate :: ActivationRequest -> AC.Action DB RequestResult
 activate (ActivationRequest p) = AC.logIfError'P $ do
     AC.logger'P DEBUG . ("route activate:" <>) . cs . Aeson.encodePretty $ ActivationRequest p
+    config <- AC.getConfig'P
     ctok        :: ConfirmationToken             <- confirmationTokenFromPath p
     (uid, stok) :: (UserId, ThentosSessionToken) <- A.confirmNewUser ctok
-    return $ RequestSuccess (userIdToPath uid) stok
+    return $ RequestSuccess (userIdToPath config uid) stok
 
 
 login :: LoginRequest -> AC.Action DB RequestResult
 login r = AC.logIfError'P $ do
     AC.logger'P DEBUG "/login/"
+    config <- AC.getConfig'P
     (uid, stok) <- case r of
         LoginByName  uname pass  -> A.startThentosSessionByUserName uname pass
         LoginByEmail email pass -> A.startThentosSessionByUserEmail email pass
-    return $ RequestSuccess (userIdToPath uid) stok
+    return $ RequestSuccess (userIdToPath config uid) stok
 
 
 -- * aux
 
-userIdToPath :: UserId -> Path
-userIdToPath (UserId i) = Path . cs $ (printf "/principals/users/%7.7i" i :: String)
+userIdToPath :: ThentosConfig -> UserId -> Path
+userIdToPath config (UserId i) = Path $ domain <> userpath
+  where
+    domain   = cs $ exposeUrl beHttp
+    userpath = cs (printf "principals/users/%7.7i" i :: String)
+    beHttp   = case config >>. (Proxy :: Proxy '["backend"]) of
+                    Nothing -> error "userIdToPath: backend not configured!"
+                    Just v -> Tagged v
 
 userIdFromPath :: Path -> AC.Action DB UserId
```

Comment:
This signature could be generalized:

```haskell
 userIdFromPath :: MonadError m => Path -> m UserId
```

Then the tests could instantiate `m` with `Maybe` instead of `Action` and would be slightly easier to write.